ddterm/shell/sd_journal.js

```diff
@@ -75,7 +79,17 @@ export function sd_journal_stream_fd(identifier, priority = LOG_INFO, level_pref
     }
 
     try {
-        GLib.unix_set_fd_nonblocking(fd, false);
+        // BEGIN ESM
+        if (major > 45) {
```

Comment:
Maybe just wrap `Gi.require()` in `try {} catch`? It seems that the exception thrown by `Gi.require()` isn't `GLib.Error`, so it should be easy to distinguish from `set_fd_nonblocking()` failure.

Reply:
I'm not sure I am a fan of having multiple nested `try {} catch` re-throwing errors. I think it adds complexity compared to the shell version check, as I don't see a real reason to make it dynamic if there is no need for more error handling.

Reply:
If you insist on version check, you should check the version of GLib, not Shell. https://github.com/GNOME/glib/commit/5021d002abe55e4d09e7e83b3ffe3a8cf1a074ac

Reply:
> If you insist on version check, you should check the version of GLib, not Shell. [GNOME/glib@5021d00](https://github.com/GNOME/glib/commit/5021d002abe55e4d09e7e83b3ffe3a8cf1a074ac)

Done in my last push. :+1: